### PR TITLE
Port to setuptools, PEP517

### PR DESCRIPTION
**Pull request by parona-source**
Would address this https://bugs.gentoo.org/841134. 

And allows using python3.12 now that it has a release and has completely removed distutils.

Decided to not touch modify setup.py further as options for building and installing gettext hasn't improved much. Looked at https://github.com/breezy-team/setuptools-gettext, but it didn't install them correctly (and I wasn't keen on troubleshooting it further).

Pyliblzma could be removed or ported away from as that had its last release in 2010. https://docs.python.org/3/library/lzma.html?

And adding a maintainer to the package metadata could be an idea.

Before:
```
Archive:  old_dist/elogv-0.8.0-py3-none-any.whl
  Length      Date    Time    Name
---------  ---------- -----   ----
     3910  10-05-2023 04:47   elogv-0.8.0.data/data/share/locale/de/LC_MESSAGES/elogv.mo
     4148  10-05-2023 04:47   elogv-0.8.0.data/data/share/locale/es/LC_MESSAGES/elogv.mo
     4022  10-05-2023 04:47   elogv-0.8.0.data/data/share/locale/it/LC_MESSAGES/elogv.mo
     3943  10-05-2023 04:47   elogv-0.8.0.data/data/share/locale/pl/LC_MESSAGES/elogv.mo
     2025  10-05-2023 04:02   elogv-0.8.0.data/data/share/man/de/man1/elogv.1.de
     2355  10-05-2023 04:02   elogv-0.8.0.data/data/share/man/es/man1/elogv.1.es
     2225  10-05-2023 04:02   elogv-0.8.0.data/data/share/man/it/man1/elogv.1.it
     1886  10-04-2023 18:21   elogv-0.8.0.data/data/share/man/man1/elogv.1
     2109  10-05-2023 04:02   elogv-0.8.0.data/data/share/man/pl/man1/elogv.1.pl
    24684  10-05-2023 04:47   elogv-0.8.0.data/scripts/elogv
      242  10-05-2023 04:47   elogv-0.8.0.dist-info/METADATA
       92  10-05-2023 04:47   elogv-0.8.0.dist-info/WHEEL
        1  10-05-2023 04:47   elogv-0.8.0.dist-info/top_level.txt
     1365  10-05-2023 04:47   elogv-0.8.0.dist-info/RECORD
---------                     -------
    53007                     14 files
```

After:
```
Archive:  dist/elogv-0.8.0-py3-none-any.whl
  Length      Date    Time    Name
---------  ---------- -----   ----
     3910  10-05-2023 04:47   elogv-0.8.0.data/data/share/locale/de/LC_MESSAGES/elogv.mo
     4148  10-05-2023 04:47   elogv-0.8.0.data/data/share/locale/es/LC_MESSAGES/elogv.mo
     4022  10-05-2023 04:47   elogv-0.8.0.data/data/share/locale/it/LC_MESSAGES/elogv.mo
     3943  10-05-2023 04:47   elogv-0.8.0.data/data/share/locale/pl/LC_MESSAGES/elogv.mo
     2025  10-05-2023 04:02   elogv-0.8.0.data/data/share/man/de/man1/elogv.1.de
     2355  10-05-2023 04:02   elogv-0.8.0.data/data/share/man/es/man1/elogv.1.es
     2225  10-05-2023 04:02   elogv-0.8.0.data/data/share/man/it/man1/elogv.1.it
     1886  10-04-2023 18:21   elogv-0.8.0.data/data/share/man/man1/elogv.1
     2109  10-05-2023 04:02   elogv-0.8.0.data/data/share/man/pl/man1/elogv.1.pl
    24684  10-05-2023 04:47   elogv-0.8.0.data/scripts/elogv
      270  10-05-2023 04:47   elogv-0.8.0.dist-info/METADATA
       92  10-05-2023 04:47   elogv-0.8.0.dist-info/WHEEL
        1  10-05-2023 04:47   elogv-0.8.0.dist-info/top_level.txt
     1365  10-05-2023 04:47   elogv-0.8.0.dist-info/RECORD
---------                     -------
    53035                     14 files
```

### CODE REVIEW

**Review by hartwork — CHANGES_REQUESTED** [review]
Hi @parona-source,

that's valuable and very clean work — cool! :+1: :+1: 

I have four things for this pull request on my mind:
- This repository is only a mirror, so I will need to merge using `git push` rather than the merge button, just so you know. I may need to add line `Signed-off-by: Sebastian Pipping <sping@gentoo.org>` to every commit for the server to accept my push, will see.
- I'm getting `WARNING pyproject.toml does not contain a tool.setuptools_scm section` when running `python -m build` locally. Should this be addressed?
- pyliblzma turns out…

  - to be a bit more than an optional dependency: if .xy files are found, elog will exit with an error
  - to not be packaged in Gentoo
  - to be long dead according to https://pypi.org/project/pyliblzma/#history

  …so it would be great to be replaced (outside of this pull request), just felt like worth mentioning.
- One single comment on details below

**Comment by hartwork** [comment]
> Pyliblzma could be removed or ported away from as that had its last release in 2010. https://docs.python.org/3/library/lzma.html?

@parona-source only seen that^^ now, sounds great!

**Comment by parona-source** [comment]
> I'm getting WARNING pyproject.toml does not contain a tool.setuptools_scm section when running python -m build locally. Should this be addressed?

I don't see this on my end (python -m build is done in an isolated venv for me). But I can get it trigger if I install setuptools_scm specifically and disable the isolation.

I don't think it matters as elogv doesn't use setuptools_scm features (hardcoded version).

<details>

```
$ python -m build 
* Creating venv isolated environment...
* Installing packages in isolated environment... (setuptools)
* Getting build dependencies for sdist...
running egg_info
writing elogv.egg-info/PKG-INFO
writing dependency_links to elogv.egg-info/dependency_links.txt
writing requirements to elogv.egg-info/requires.txt
writing top-level names to elogv.egg-info/top_level.txt
reading manifest file 'elogv.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'elogv.egg-info/SOURCES.txt'
* Building sdist...
running sdist
running egg_info
writing elogv.egg-info/PKG-INFO
writing dependency_links to elogv.egg-info/dependency_links.txt
writing requirements to elogv.egg-info/requires.txt
writing top-level names to elogv.egg-info/top_level.txt
reading manifest file 'elogv.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'elogv.egg-info/SOURCES.txt'
running check
creating elogv-0.8.0
creating elogv-0.8.0/elogv.egg-info
creating elogv-0.8.0/man
creating elogv-0.8.0/po
copying files to elogv-0.8.0...
copying ChangeLog -> elogv-0.8.0
copying ChangeLog.old -> elogv-0.8.0
copying INSTALL -> elogv-0.8.0
copying MANIFEST.in -> elogv-0.8.0
copying Makefile -> elogv-0.8.0
copying README -> elogv-0.8.0
copying elogv -> elogv-0.8.0
copying pyproject.toml -> elogv-0.8.0
copying setup.py -> elogv-0.8.0
copying elogv.egg-info/PKG-INFO -> elogv-0.8.0/elogv.egg-info
copying elogv.egg-info/SOURCES.txt -> elogv-0.8.0/elogv.egg-info
copying elogv.egg-info/dependency_links.txt -> elogv-0.8.0/elogv.egg-info
copying elogv.egg-info/requires.txt -> elogv-0.8.0/elogv.egg-info
copying elogv.egg-info/top_level.txt -> elogv-0.8.0/elogv.egg-info
copying man/elogv.1 -> elogv-0.8.0/man
copying man/elogv.1.de -> elogv-0.8.0/man
copying man/elogv.1.es -> elogv-0.8.0/man
copying man/elogv.1.it -> elogv-0.8.0/man
copying man/elogv.1.pl -> elogv-0.8.0/man
copying po/de.po -> elogv-0.8.0/po
copying po/elogv.pot -> elogv-0.8.0/po
copying po/es.po -> elogv-0.8.0/po
copying po/it.po -> elogv-0.8.0/po
copying po/pl.po -> elogv-0.8.0/po
Writing elogv-0.8.0/setup.cfg
Creating tar archive
removing 'elogv-0.8.0' (and everything under it)
* Building wheel from sdist
* Creating venv isolated environment...
* Installing packages in isolated environment... (setuptools)
* Getting build dependencies for wheel...
running egg_info
writing elogv.egg-info/PKG-INFO
writing dependency_links to elogv.egg-info/dependency_links.txt
writing requirements to elogv.egg-info/requires.txt
writing top-level names to elogv.egg-info/top_level.txt
reading manifest file 'elogv.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'elogv.egg-info/SOURCES.txt'
* Installing packages in isolated environment... (wheel)
* Building wheel...
running bdist_wheel
running build
running build_scripts
creating build
creating build/scripts-3.12
copying and adjusting elogv -> build/scripts-3.12
changing mode of build/scripts-3.12/elogv from 644 to 755
running build_messages
creating build/po
Building catalog po/de.po > build/po/de.gmo
Building catalog po/es.po > build/po/es.gmo
Building catalog po/it.po > build/po/it.gmo
Building catalog po/pl.po > build/po/pl.gmo
installing to build/bdist.linux-x86_64/wheel
running install
running install_egg_info
running egg_info
writing elogv.egg-info/PKG-INFO
writing dependency_links to elogv.egg-info/dependency_links.txt
writing requirements to elogv.egg-info/requires.txt
writing top-level names to elogv.egg-info/top_level.txt
reading manifest file 'elogv.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'elogv.egg-info/SOURCES.txt'
Copying elogv.egg-info to build/bdist.linux-x86_64/wheel/elogv-0.8.0-py3.12.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/scripts
copying build/scripts-3.12/elogv -> build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/scripts
changing mode of build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/scripts/elogv to 755
running install_messages
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/de
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/de/LC_MESSAGES
copying build/po/de.gmo -> build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/de/LC_MESSAGES/elogv.mo
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/es
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/es/LC_MESSAGES
copying build/po/es.gmo -> build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/es/LC_MESSAGES/elogv.mo
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/it
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/it/LC_MESSAGES
copying build/po/it.gmo -> build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/it/LC_MESSAGES/elogv.mo
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/pl
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/pl/LC_MESSAGES
copying build/po/pl.gmo -> build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/locale/pl/LC_MESSAGES/elogv.mo
running install_manpages
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/man1
copying man/elogv.1 -> build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/man1
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/de
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/de/man1
copying man/elogv.1.de -> build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/de/man1
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/es
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/es/man1
copying man/elogv.1.es -> build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/es/man1
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/it
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/it/man1
copying man/elogv.1.it -> build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/it/man1
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/pl
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/pl/man1
copying man/elogv.1.pl -> build/bdist.linux-x86_64/wheel/elogv-0.8.0.data/data/share/man/pl/man1
creating build/bdist.linux-x86_64/wheel/elogv-0.8.0.dist-info/WHEEL
creating '/home/ask/sources/elogv/dist/.tmp-s4oxd88t/elogv-0.8.0-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'elogv-0.8.0.data/data/share/locale/de/LC_MESSAGES/elogv.mo'
adding 'elogv-0.8.0.data/data/share/locale/es/LC_MESSAGES/elogv.mo'
adding 'elogv-0.8.0.data/data/share/locale/it/LC_MESSAGES/elogv.mo'
adding 'elogv-0.8.0.data/data/share/locale/pl/LC_MESSAGES/elogv.mo'
adding 'elogv-0.8.0.data/data/share/man/de/man1/elogv.1.de'
adding 'elogv-0.8.0.data/data/share/man/es/man1/elogv.1.es'
adding 'elogv-0.8.0.data/data/share/man/it/man1/elogv.1.it'
adding 'elogv-0.8.0.data/data/share/man/man1/elogv.1'
adding 'elogv-0.8.0.data/data/share/man/pl/man1/elogv.1.pl'
adding 'elogv-0.8.0.data/scripts/elogv'
adding 'elogv-0.8.0.dist-info/METADATA'
adding 'elogv-0.8.0.dist-info/WHEEL'
adding 'elogv-0.8.0.dist-info/top_level.txt'
adding 'elogv-0.8.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Successfully built elogv-0.8.0.tar.gz and elogv-0.8.0-py3-none-any.whl
```
</details>

File: setup.py [hunk 1]
```diff
@@ -251,14 +204,7 @@ def run(self):
 
 man_pages = glob("man/*")
 
-setup(name="elogv",
-      version="0.8.0",
-      author="Luca Marturana",
-      author_email="lucamarturana@gmail.com",
-      license="GPL-2",
-      description="Curses based utility to view elogs created by Portage",
-      url="https://gitweb.gentoo.org/proj/elogv.git/",
```
[the remark this answers is not in the document]
Comment (hartwork) [reply setup.py:260]: @parona-source I think homepage will do for a start, thanks for the link and the fix!

**Comment by parona-source** [comment]
Started to look at lzma as it appeared to be quick process. But I got two oddities, 1. portage doesn't appear to have mechanism to compress elog files 2. manually compressing the file find areas where decompression is ignored completely https://github.com/gentoo/elogv/blob/master/elogv#L531. 

So this would appear to either a case for refactoring or dropping support for decompression.

**Comment by hartwork** [comment]
@parona-source interesting! My guess would be `app-admin/logrotate`. If it's half working without crashing (guessing from the linked code falling back to `einfo`) maybe fixing it will benefit some users, if you feel like it :shrug:

**Comment by parona-source** [comment]
```
Traceback (most recent call last):
  File "/home/ask/.virtualenvs/elogv/bin/elogv", line 681, in <module>
    curses.wrapper(ElogViewer)
  File "/usr/lib/python3.12/curses/__init__.py", line 94, in wrapper
    return func(stdscr, *args, **kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ask/.virtualenvs/elogv/bin/elogv", line 227, in __init__
    self.init_screen()
  File "/home/ask/.virtualenvs/elogv/bin/elogv", line 364, in init_screen
    self.fill_file_pad()
  File "/home/ask/.virtualenvs/elogv/bin/elogv", line 470, in fill_file_pad
    self.packages.append( (filepath, cat, pn, date, self.get_class(filepath)) )
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ask/.virtualenvs/elogv/bin/elogv", line 524, in get_class
    classes = re.findall("LOG:|INFO:|WARN:|ERROR:", f.read())
                                                    ^^^^^^^^
  File "<frozen codecs>", line 322, in decode
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xef in position 12: invalid continuation byte
```

It doesn't catch the exception so it appears to be completely broken. I guess nobody has used logrotate like that for it and caught this issue.

**Comment by hartwork** [comment]
@parona-source interesting. I'm happy to review whatever you feel comfortable and responsible to patch.

**Pull request merged**